### DISCUSSION TITLE
Deepgram SDK 3 using Quart

### DISCUSSION THREAD

**Discussion by cardinalfan1** [deepgram/community#516] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

I am trying to adapt this tutorial ( [link](https://deepgram.com/learn/live-transcription-quart) ) the SDK 3. I have a Quart setup for my backend. Here is the code:

`async def connect_deepgram_live():
    options = LiveOptions(
        model="nova-2",
        interim_results=False,
        language="en-US",
    )
    
    print("CONNECTING TO DEEPGRAM LIVE")
    dg_connection = deepgram.listen.asynclive.v("1")
    
    async def on_message(self, result, **kwargs):
        sentence = result.channel.alternatives[0].transcript
        if len(sentence) == 0:
            return
        logger.info(sentence)

    async def on_metadata(self, metadata, **kwargs):
        logger.info(f"{metadata}")

    #def on_utterance_end(self, utterance_end, **kwargs):
    #    logger.info(f"{utterance_end}")

    async def on_error(self, error, **kwargs):
        logger.error(f"{error}")

    dg_connection.on(LiveTranscriptionEvents.Transcript, on_message)
    dg_connection.on(LiveTranscriptionEvents.Metadata, on_metadata)
    #dg_connection.on(LiveTranscriptionEvents.UtteranceEnd, on_utterance_end)
    dg_connection.on(LiveTranscriptionEvents.Error, on_error)

    # connect to websocket
    await dg_connection.start(options)

    return dg_connection

async def disconnect_deepgram_live(dg_connection):
    if dg_connection:
        print("DISCONNECTING FROM DEEPGRAM LIVE")
        await dg_connection.finish()

async def send_deepgram_live(data, dg_connection):
    if dg_connection:
        print("SENDING DATA TO DEEPGRAM")
        await dg_connection.send(data)

@app.websocket('/dictation')
async def dictation_socket():
    await websocket.accept()
    dg_connection = None
    try:
        while True:
            #try:
            data = await websocket.receive()

            if not dg_connection:
                dg_connection = await connect_deepgram_live()

            try:
                js = json.loads(data)
                if js.get("action") == "stop_dictation":
                    print("GOT STOP SIGNAL...")
                    await disconnect_deepgram_live(dg_connection)
                    dg_connection = None
            except UnicodeDecodeError:
                # Assuming data is non-JSON (e.g., binary audio data)
                if dg_connection:
                    try:
                        print("GOT DATA")
                        await send_deepgram_live(data, dg_connection)
                    except Exception as e:
                        print(e)
                    
    #except Exception as e:
    #    print(f'Error during WebSocket communication: {e}')
    finally:
        if dg_connection:
            await disconnect_deepgram_live(dg_connection)
            dg_connection = None`

The backend is able to stream a few packets to deepgram but always disconnects quickly: `MetadataResponse(type='Metadata', transaction_key='deprecated', request_id='ea318d5a-fb38-427e-9519-35fb6dd44b86', sha256='f855ec29096c78321bb664ffb390e1291218f10e8ccb7dd9f26532e764cfd34f', created='2024-01-06T15:02:12.793Z', duration=0.0599375, channels=1, models=['c9d824fb-4090-43a0-8e06-6d87aa976780'], model_info={'c9d824fb-4090-43a0-8e06-6d87aa976780': ModelInfo(name='2-general-nova', version='2023-11-14.6411', arch='nova-2')})`

It's not an issue between my client and server, because if i comment out the send command, it will keep transmitting the data correctly.

Also, interestingly, if I specify the encoding as "linear16" it will not shut down the connection.... although I am using Chrome and it should be submitting as WebM and I should not have to specify the encoding.

Thanks for your help

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
this was answered in Discord
https://discord.com/channels/1108042150941294664/1192992302080213122
